components/SearchBar/index.js

```diff
@@ -0,0 +1,128 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { useState } from "react";
+import GetNotes from "../GetNotes";
+import { cards } from "../../lib/data";
+import useLocalStorageState from "use-local-storage";
+
+export default function SearchBar({}) {
+  const router = useRouter();
+  const id = router ? router.query.id : null;
+  const [input, setInput] = useState("");
+  const [searchOptionSelect, setSearchOptionSelect] = useState("resultsByName");
+  const notes = GetNotes();
+  const [searchResults, setSearchResults] = useLocalStorageState(
+    "searchquery",
+    { defaultValue: [] }
+  );
+
+  function searchForInput(event) {
+    const { value } = event.target;
+    setInput(value);
+    let results = "";
+    let queryWords = value
+      .split(" ")
+      .filter((queryWord) => queryWord.length > 0);
+    if (value) {
+      if (searchOptionSelect === "resultsByName") {
```

Comment:
These if Statements are very repetitive. Im not understanding this code fully, but I am sure you can write just one function and give it the Search option as a prop.

Reply:
i will look inside that issue, if i can get it more compact through properties. wasn't thinking about it at that moment! was just happy this - pretty big code (sorry for that) - was working as intended :)